de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlDeliveryOrderRepository.java

```diff
@@ -0,0 +1,380 @@
+								.alpha2(firstOrder.getDHL_Shipper_CountryISO2Code())
+								.alpha3(firstOrder.getDHL_Shipper_CountryISO3Code())
+								.build())
+						.build())
+				.pickupDate(PickupDate.builder()
+						.date(LocalDate.parse(firstOrder.getDHL_ShipmentDate(), DateTimeFormatter.ISO_LOCAL_DATE))
+						.build())
+				// other
+				.customerReference(firstOrder.getCustomerReference())
+				.serviceType(DhlServiceType.valueOf(firstOrder.getDHL_Product()))
+				.deliveryPositions(constructDeliveryPositions(firstOrder, packageIds))
+				.shipperId(firstOrder.getM_Shipper_ID())
+				.shipperTransportationId(firstOrder.getM_ShipperTransportation_ID())
+				.customDeliveryData(DhlCustomDeliveryData.builder()
+						.details(dhlCustomDeliveryDataDetail)
+						.build())
+				.build();
+
+		return doFromPO;
+	}
+
+	private static List<I_DHL_ShipmentOrder> getAllShipmentOrdersForRequest(final int requestId)
+	{
+		return Services.get(IQueryBL.class)
+				.createQueryBuilderOutOfTrx(I_DHL_ShipmentOrder.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_DHL_ShipmentOrder.COLUMNNAME_DHL_ShipmentOrderRequest_ID, requestId)
+				.create()
+				.list();
+	}
+
+	@NonNull
+	private Iterable<? extends DeliveryPosition> constructDeliveryPositions(
+			@NonNull final I_DHL_ShipmentOrder firstOrder,
+			@NonNull final ImmutableSet<Integer> packageIds)
+	{
+		final DeliveryPosition singleDeliveryPosition = DeliveryPosition.builder()
+				.packageDimensions(PackageDimensions.builder()
+						.widthInCM(firstOrder.getDHL_WidthInCm())
+						.lengthInCM(firstOrder.getDHL_LengthInCm())
+						.heightInCM(firstOrder.getDHL_HeightInCm())
+						.build())
+				.grossWeightKg(firstOrder.getDHL_WeightInKg().intValue())
+				.numberOfPackages(packageIds.size())
+				.packageIds(packageIds)
+				.build();
+
+		return Collections.singleton(singleDeliveryPosition);
+	}
+
+	/**
+	 * Persists the shipper-dependant DeliveryOrder details
+	 * <p>
+	 * todo: https://chat.metasfresh.org/metasfresh/pl/z7pahtuci3b35naios3ha16tnh
+	 * <p>
+	 * keep in sync with {@link #toDeliveryOrderFromPO(I_DHL_ShipmentOrderRequest)}
+	 */
+	@NonNull
+	private I_DHL_ShipmentOrderRequest toCreateShipmentOrderRequestPO(@NonNull final DeliveryOrder deliveryOrder)
+	{
+		final I_DHL_ShipmentOrderRequest shipmentOrderRequest = InterfaceWrapperHelper.newInstance(I_DHL_ShipmentOrderRequest.class);
+		InterfaceWrapperHelper.save(shipmentOrderRequest);
+
+		// maybe this will be removed in the future, but for now it simplifies the PO deserialisation implementation and other implementation details dramatically, ref: constructDeliveryPositions()
+		// therefore please ignore the for loops over `deliveryOrder.getDeliveryPositions()` as they don't help at all
+		Check.errorIf(deliveryOrder.getDeliveryPositions().size() != 1,
+				"The DHL implementation needs to always create DeliveryOrders with exactly 1 DeliveryPosition; deliveryOrder={}",
+				deliveryOrder);
+
+		//		int totalPackages = 0;
```

Comment:
pls clean up the commented-out code in here, or add more comments ;-) so it becomes clear how they might be are useful in the future